logstash-core/src/main/java/org/logstash/instrument/witness/ReloadWitness.java

```diff
@@ -99,6 +113,50 @@ public void lastFailureTimestamp(JrubyTimestampExtLibrary.RubyTimestamp timestam
         lastFailureTimestamp.set(timestamp);
     }
 
+    @Override
+    public void genJson(JsonGenerator gen, SerializerProvider provider) throws IOException {
+        new Serializer().innerSerialize(this, gen, provider);
+    }
+
+    /**
+     * The Jackson serializer.
+     */
+    public static class Serializer extends StdSerializer<ReloadWitness> {
+
+        /**
+         * Default constructor - required for Jackson
+         */
+        public Serializer() {
+            this(ReloadWitness.class);
+        }
+
+        /**
+         * Constructor
+         *
+         * @param t the type to serialize
+         */
+        protected Serializer(Class<ReloadWitness> t) {
+            super(t);
+        }
+
+        @Override
+        public void serialize(ReloadWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeStartObject();
+            innerSerialize(witness, gen, provider);
+            gen.writeEndObject();
+        }
+
+        void innerSerialize(ReloadWitness witness, JsonGenerator gen, SerializerProvider provider) throws IOException {
+            gen.writeObjectFieldStart(ReloadWitness.KEY);
+            witness.lastError.genJson(gen, provider);
+            MetricSerializer.Get.longSerializer(gen).serialize(witness.success);
```

Comment:
@jakelandis I meant caching the return of `MetricSerializer.Get.longSerializer(gen)` into a variable, since this is a stateless thing isn't it? To avoid the repeated call to `longSerializer()` and `timestampSerializer()`?